### PR TITLE
[click] Update to use openstore v3 api

### PR DESCRIPTION
**Pull request by mariogrip**

### CODE REVIEW

**Comment by mariogrip** [comment]
seems to work fine for me

File: plugins/system-update/click/apiclient_impl.cpp [hunk 1]
```diff
@@ -78,33 +83,26 @@ void ApiClientImpl::requestMetadata(const QUrl &url,
...
-    QUrl u(url);
-    u.setQuery(query);
-    QNetworkRequest request;
-    request.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
-    request.setUrl(u);
-    request.setOriginatingObject(this);
-    request.setAttribute(QNetworkRequest::User, "metadata-request");
-
-    initializeReply(m_nam->post(request, content));
+    m_hasErrors = false;
```
Comment (mardy) [inline plugins/system-update/click/apiclient_impl.cpp]: please also set `m_requests` to 0 here, it's safer. Maybe something like this:

    if (m_requests != 0) {
        qCritical() << Q_FUNC_INFO << "Has still some requests active:" << m_requests;
        m_requests = 0;
    }

File: plugins/system-update/click/apiclient_impl.cpp [hunk 2]
```diff
@@ -78,33 +83,26 @@ void ApiClientImpl::requestMetadata(const QUrl &url,
...
+    foreach (const auto &package, packages) {
+        QString rawUrl(Helpers::clickMetadataUrl());
+        rawUrl.append(package);
+        QUrl url(rawUrl);
+        QUrlQuery query(url);
+
+        qWarning() << url;
```
Comment (mardy) [inline plugins/system-update/click/apiclient_impl.cpp]: I suggest printing `u` a few lines below, so we also see the query parameters.

**Review by mardy — CHANGES_REQUESTED** [review]
Just a couple of minor remarks. I still have to test it, will do that very soon :-)

File: plugins/system-update/click/apiclient_impl.cpp (hunk 2 above)
Comment (mariogrip) [reply plugins/system-update/click/apiclient_impl.cpp]: Ups, was not meant to be included

**Review by mardy — APPROVED** [review]
Works like a charm!

**Comment by UniversalSuperBox** [comment]
Don't forget translations!

**Review by UniversalSuperBox — APPROVED** [review]
Eh, never mind. I'll just pull them in to #149

**Pull request merged**